version: "3.8"

services:
  app:
    build:
      context: .  
      dockerfile: Dockerfile.app 
    container_name: my-app
    depends_on:
      - postgresql  
    environment:
      - DATABASE_URL=postgres://my_user:my_password@postgres:5432/my_database  
    ports:
      - "8080:80" 
    networks:
      - app-network

  postgresql:
    build:
      context: .  
      dockerfile: Dockerfile.db 
    container_name: postgresql
    environment:
      - POSTGRES_USER=my_user
      - POSTGRES_PASSWORD=my_password
      - POSTGRES_DB=my_database
    volumes:
      - ./data:/var/lib/postgresql/data  
    ports:
      - "5432:5432"
    networks:
      - app-network
    healthcheck:
      test: ["CMD", "pg_isready", "-U", "my_user", "-d", "my_database"]
      interval: 10s
      retries: 5

networks:
  app-network:  
    driver: bridge